src/frontend/metrics/AppDiagnosticMetrics.ts

```diff
@@ -0,0 +1,200 @@
+import * as NetInfo from '@react-native-community/netinfo';
+import * as Sentry from '@sentry/react-native';
+import * as Application from 'expo-application';
+import {getLastKnownPositionAsync} from 'expo-location';
+import pProps from 'p-props';
+import {AppState, Platform} from 'react-native';
+import {storage} from '../hooks/persistedState/createPersistedState';
+import {
+  getDeviceLanguageTag,
+  usePersistedLocale,
+} from '../hooks/persistedState/usePersistedLocale';
+import {MINUTE_MS, formatIsoUtc} from '../lib/date';
+import {first} from '../lib/first';
+import {maybeJsonParse} from '../lib/maybeJsonParse';
+import {OneAtATimeQueue} from '../lib/OneAtATimeQueue';
+import {setIfNotNull} from '../lib/setIfNotNull';
+import {
+  type AppDiagnosticMetricsQueue,
+  type AppDiagnosticMetricsReport,
+  getRequestDatas,
+  hasReportForToday,
+  truncateReportsByTime,
+  updateQueueHighWatermark,
+} from './AppDiagnosticMetricsQueue';
+import {getMetricsDeviceId} from './getMetricsDeviceId';
+import {getMetricsRequestInfo} from './getMetricsRequestInfo';
+import {getMonthlyHash} from './getMonthlyHash';
+import {positionToCountries} from './positionToCountries';
+import {sendMetricsData} from './sendMetricsData';
+
+const STORAGE_KEY = 'AppDiagnosticMetricsQueue';
+const CHECK_INTERVAL = 5 * MINUTE_MS;
+
+function loadQueueFromStorage(): AppDiagnosticMetricsQueue {
+  const storedString = storage.getString(STORAGE_KEY);
+  if (!storedString) return {reports: []};
+
+  const stored = maybeJsonParse(storedString);
+  if (!stored) return {reports: []};
+
+  return stored as AppDiagnosticMetricsQueue;
```

Comment:
This cast is not type-safe. The stored value could be of a type other than `AppDiagnosticMetricsQueue`. I considered writing a bunch of code to check the type (possibly using something like [Zod](https://www.npmjs.com/package/zod)) but I chose not to.